From c1787a2abd008e86593f1ed83a501eeb8985a6d5 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 20 Nov 2017 15:19:41 +0200
Subject: [PATCH 2176/2241] fix: net: mvpp2x: fix sram.byte array
 initialization

Fix incorrect index-calculating causing for memory-overwrite
in array sram.byte[index] access and
reported by compiler as "array subscript is above array bounds"

sram.byte[] is used for PRS bit-converting

Change-Id: I75032e7797ce07d0069c6ae75f1609039b7942e5
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/46539
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
index 9e423fa..4628728 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
@@ -4566,16 +4566,16 @@ int mv_pp2x_prs_sw_sram_offset_get(struct mv_pp2x_prs_entry *pe,
 		MVPP2_PRS_SRAM_UDF_OFFS)] >>
 		(MVPP2_PRS_SRAM_UDF_OFFS % 8)) & 0x7f;
 	*offset |= (pe->sram.byte[
-		    SRAM_BIT_TO_BYTE(MVPP2_PRS_SRAM_UDF_OFFS) +
-		    SRAM_BIT_TO_BYTE(MVPP2_PRS_SRAM_UDF_OFFS)] <<
+		    SRAM_BIT_TO_BYTE(MVPP2_PRS_SRAM_UDF_OFFS +
+		    MVPP2_PRS_SRAM_UDF_BITS)] <<
 		    (8 - (MVPP2_PRS_SRAM_UDF_OFFS % 8))) & 0x80;
 
 	*op = (pe->sram.byte[SRAM_BIT_TO_BYTE(
 		MVPP2_PRS_SRAM_OP_SEL_SHIFT_OFFS)] >>
 		(MVPP2_PRS_SRAM_OP_SEL_SHIFT_OFFS % 8)) & 0x7;
-	*op |= (pe->sram.byte[HW_BYTE_OFFS(
-		SRAM_BIT_TO_BYTE(MVPP2_PRS_SRAM_OP_SEL_SHIFT_OFFS) +
-		SRAM_BIT_TO_BYTE(MVPP2_PRS_SRAM_OP_SEL_SHIFT_OFFS))] <<
+	*op |= (pe->sram.byte[SRAM_BIT_TO_BYTE(
+		MVPP2_PRS_SRAM_OP_SEL_SHIFT_OFFS +
+		MVPP2_PRS_SRAM_OP_SEL_SHIFT_BITS)] <<
 		(8 - (MVPP2_PRS_SRAM_OP_SEL_SHIFT_OFFS % 8))) & 0x18;
 
 	/* if signed bit is tes */
-- 
2.7.4

